### PR TITLE
Learn Search Redesign

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Learn Search Redesign 

search

### CODE REVIEW

**Comment by rtibbles** [comment]
I think you want to target develop with this.

**Comment by rtibbles** [comment]
Hrm.. could you rebase to develop and retarget?

File: kolibri/plugins/learn/assets/src/views/index.vue [hunk 1]
```diff
@@ -2,14 +2,38 @@
...
+
+      <form
+        @submit.prevent="search"
+        class="search-box">
+        <input
+          type="search"
+          :placeholder="$tr('search')"
+          v-model="searchQuery"
+          class="search-input">
         <ui-icon-button
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/index.vue:14]: Should probably swap this out with our wrapped component?

File: kolibri/plugins/learn/assets/src/views/index.vue [hunk 2]
```diff
@@ -81,6 +105,13 @@
       'exam-list': require('./exam-list'),
       'exam-page': require('./exam-page'),
       'total-points': require('./total-points'),
+      'ui-icon-button': require('keen-ui/src/UiIconButton'),
+      'ui-icon': require('keen-ui/src/UiIcon'),
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/index.vue:109]: I thought we were leaning away from using keen elements directly?

File: kolibri/core/assets/src/views/tab-button/index.vue [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<template>
+
+  <li>
+    <button
```
Comment (DXCanas) [inline kolibri/core/assets/src/views/tab-button/index.vue:4]: What's the difference between this and a `tab-link`? One changes route and another doesn't?

i stand by my suggestion of [scoped slots](https://vuejs.org/v2/guide/components.html#Scoped-Slots)

File: kolibri/core/assets/src/views/tab-link/index.vue [hunk 4]
```diff
@@ -32,11 +31,11 @@
       },
       title: {
         type: String,
-        required: () => this.type === 'title' || this.type === 'icon-and-title',
+        required: false,
```
Comment (DXCanas) [inline kolibri/core/assets/src/views/tab-link/index.vue:34]: [:smirk:] (https://github.com/learningequality/kolibri/pull/1475#issuecomment-301216674)

File: kolibri/core/assets/src/views/textbox/index.vue [hunk 5]
```diff
@@ -85,6 +85,11 @@
         this.$refs.textbox.reset();
       },
     },
+    watch: {
+      value(val) {
+        this.currentText = val;
```
Comment (DXCanas) [inline kolibri/core/assets/src/views/textbox/index.vue:90]: Makes me sad. But it's gotta happen.

File: kolibri/plugins/learn/assets/src/views/index.vue [hunk 6]
```diff
@@ -168,6 +207,11 @@
         return { name: PageNames.EXAM_LIST, params: { channel_id: this.channelId } };
       },
     },
+    watch: {
+      searchTerm(val) {
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/index.vue:211]: Might be better to make this a computed prop

File: kolibri/plugins/learn/assets/src/views/search-page/index.vue [hunk 7]
```diff
@@ -1,149 +1,193 @@
...
+          type="icon-and-title"
+          icon="widgets"
+          :title="$tr('html5', { num: html5.length } )"
+          :selected="filter === contentNodeKinds.HTML5"
+          @click="filter = contentNodeKinds.HTML5"
+        />
+      </tabs>
+
+      <p v-if="filteredResults.length === 0">
+        <template v-if="filter === 'all'">{{ $tr('noContent', { searchTerm }) }}</template>
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/search-page/index.vue]: why not just make all of these options a computed prop that returns a translated string?

File: kolibri/plugins/learn/assets/src/views/search-page/index.vue [hunk 8]
```diff
@@ -1,149 +1,193 @@
...
+        <template v-else-if="filter === contentNodeKinds.TOPIC">{{ $tr('noTopics', { searchTerm }) }}</template>
+        <template v-else-if="filter === contentNodeKinds.EXERCISE">{{ $tr('noExercises', { searchTerm }) }}</template>
+        <template v-else-if="filter === contentNodeKinds.VIDEO">{{ $tr('noVideos', { searchTerm }) }}</template>
+        <template v-else-if="filter === contentNodeKinds.DOCUMENT">{{ $tr('noDocuments', { searchTerm }) }}</template>
+        <template v-else-if="filter === contentNodeKinds.HTML5">{{ $tr('noHtml5', { searchTerm }) }}</template>
+      </p>
+
+      <card-grid v-else>
+        <content-grid-item
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/search-page/index.vue]: I think this is what I'd be working on

File: kolibri/plugins/learn/assets/src/views/index.vue (hunk 1 above)
Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/views/index.vue:14]: We currently do not have a wrapper for `ui-icon-button`

File: kolibri/core/assets/src/views/tab-button/index.vue (hunk 3 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/tab-button/index.vue:4]: Yes one is a router-link while the other is just a button that fires a simple click event.

File: kolibri/plugins/learn/assets/src/views/index.vue (hunk 6 above)
Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/views/index.vue:211]: I need to watch it though. :/

Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/views/index.vue:211]: If it was a computed prop based on `this.searchTerm` why wouldn't `this.searchQuery` recompute when it changed?

File: kolibri/plugins/learn/assets/src/views/search-page/index.vue [hunk 9]
```diff
@@ -160,7 +153,30 @@
       },
       numFiltered() {
         return this.filteredResults.length;
-      }
+      },
+      noResultsMsg() {
```
Comment (DXCanas) [inline kolibri/plugins/learn/assets/src/views/search-page/index.vue]: Sorry man. I think I like [this pattern](https://github.com/christianmemije/kolibri/blob/2b143e2d9b066577b038f38f4f5b4a55354b3ed3/kolibri/plugins/learn/assets/src/views/search-page/index.vue#L140) more, in terms of legibility :/

Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/views/search-page/index.vue]: Addressed in https://github.com/learningequality/kolibri/pull/1514/commits/c0016a7e7e6238a3588d74c1951efd3c818dbd96

File: kolibri/plugins/learn/assets/src/views/index.vue (hunk 6 above)
Comment (christianmemije) [reply kolibri/plugins/learn/assets/src/views/index.vue:211]: `searchQuery` is a v-model

Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/views/index.vue:211]: Perhaps a computed prop with a setter?

```
computed: {
    searchTerm: {
        set(val) {
             this.searchQuery = val || '';
         },
         get() {
             return this.searchQuery;
         },
     },
},

**Pull request merged**